CA_DataUploaderLib/BaseSensorBox.cs

```diff
@@ -93,10 +95,37 @@ static void RegisterCustomBoardCommand(IOconfMap map, CommandHandler cmd, Dictio
                 var channelWriter = channel.Writer;
                 boardCustomCommandsReceived.Add(map.McuBoard, channel.Reader);
                 cmd.AddMultinodeCommand(
-                    "custom", 
-                    a => a.Count >= 3 && a[1] == map.BoxName, 
+                    "custom",
+                    a => a.Count >= 3 && a[1] == map.BoxName,
                     a => channelWriter.TryWrite(string.Join(' ', a.Skip(2))));
             }
+            static void RegisterReconnectBoardCommand(IOconfMap map, CommandHandler cmd, Dictionary<string, TaskCompletionSource> reconnectTasks)
+            {
+                if (!map.IsLocalBoard)
+                {//if the board is not local we register the command validation with an empty action
+                    cmd.AddMultinodeCommand("reconnect", a => a.Count == 2 && a[1] == map.BoxName, _ => { });
+                    return;
+                }
+
+                if (map.McuBoard == null)
+                {
+                    cmd.AddMultinodeCommand(
+                        "reconnect", a => a.Count == 2 && a[1] == map.BoxName,
+                        _ => { CALog.LogData(LogID.A, $"reconnect failed due to missing board: {map.BoxName}"); });
+                    return;
+                }
+
+                reconnectTasks[map.BoxName] = new TaskCompletionSource();
+                cmd.AddMultinodeCommand(
+                    "reconnect",
+                    a => a.Count == 2 && a[1] == map.BoxName,
+                    a =>
+                    {
+                        var currentTCS = reconnectTasks[map.BoxName];
+                        reconnectTasks[map.BoxName] = new TaskCompletionSource();

```

Comment:
We are writing here to a dictionary that we read from a different thread. Not if this could be or not an issue, as at least we are updating an existing entry in the dictionary

Reply:
added locking around use of dictionary